sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableService.java

```diff
@@ -45,6 +45,10 @@ interface Writer {
     CompletionStage<MutateRowResponse> writeRecord(KV<ByteString, Iterable<Mutation>> record)
         throws IOException;
 
+    /** Like above, but will not batch the record. Useful for single record retries */
+    CompletionStage<MutateRowResponse> writeRecordWithoutBatching(
```

Comment:
Consider naming - writeRecordDirectly or writeSingleRecord?

and maybe comment that this is for some specific internal usage? Also, could this affect performance?

Reply:
good call, renamed